### DISCUSSION TITLE
Packer Palettes

### DISCUSSION THREAD

**Discussion by MuhammadAttaElhamouly** [RosettaCommons/rosetta#142] [Rosetta] [answered]
Hi, 

Is it possible to select only certain amino acids for PackerPalettes and ignore all the others?
I mean not only giving priority but also defining the exact selected groups?

I checked out the Documentation and  didn't find a palette that can select certain amino acids, the whole pallet has to be included.

**Comment by vmullig** [comment, marked as answer]
PackerPalettes currently require you to specify amino acids one by one.  So
yes, you can define whatever subset you like.  Note that the PackerPalette
is meant to be a coarse, global level of control, however: the intended
workflow is that you choose all the amino acids that you may want to use
with your PackerPalette (like an artist putting paint colours on their
artist's palette), then you prohibit particular types at particular
positions with TaskOperations (like an artist masking part of a painting
before painting with red shades, for instance).  TaskOperations give you
more ways to add restrictions (using properties in addition to names, so
you could prohibit all aromatics at one position, and all chaeges types at
another).  On top of that, you can use tools like aa_composition to bias
the amino acid composition in a desired direction.

So with a PackerPalette you would say, "I think my design task might
involve norleucine and norvaline today."  With TaskOperations, you might
say, "I do not want to use any hydrophobic residue types on the protein
surface (so no norleucine or norvaline there),"  an absolute prohibition
that will ensure that the packet doesn't even consider these types at these
positions.  With aa_composition, you might say, "I want to aim for at most
two norleucine or norvaline in the core," a sort of guidance that does NOT
prevent these types from being considered.

**Reply by MuhammadAttaElhamouly** [reply]
Many thanks, this is so helpful, I'll try several times to optimize the XML script to increase its fitting.

**Comment by vmullig** [comment]
*all charged types at another.  Typing on a phone.

**Comment by vmullig** [comment]
*that the packer doesn't even consider
